src/libsync/syncengine.cpp

```diff
@@ -589,11 +589,12 @@ int SyncEngine::treewalkFile( TREE_WALK_FILE *file, bool remote )
             // This counts as a NONE for detecting if all the files on the server were changed
             _hasNoneFiles = true;
         } else if (!isDirectory) {
-            if (std::difftime(file->modtime, file->other.modtime) < 0) {
-                // We are going back on time
+            auto difftime = std::difftime(file->modtime, file->other.modtime);
+            if (difftime < -3600) {
+                // We are going back on time (more than two hours to avoid clock skew issues or DST)
                 _backInTimeFiles++;
                 qDebug() << file->path << "has a timestamp earlier than the local file";
-            } else {
+            } else if (difftime > 0) {
                 _hasForwardInTimeFiles = true;
             }
```

Comment:
Is it correct to do nothing for the -3600, 0 case? Could you add a comment about that?